fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -397,15 +325,53 @@ instance IPFS.MonadLocalIPFS Server where
 
 instance IPFS.MonadRemoteIPFS Server where
   runRemote query = do
-    peerID       <- head <$> asks ipfsRemotePeers
-    IPFS.URL url <- asks ipfsURL
-    manager      <- asks httpManager
+    clusterURLs       <- asks ipfsURLs
+    IPFS.Timeout secs <- asks ipfsTimeout
+    manager           <- asks httpManager
+
+    logDebug @Text "Running IPFS request across cluster"
+    requests <- forM clusterURLs \(IPFS.URL url) ->
+      Process.asyncFor (Seconds (Unity secs)) do
+        runClientM query $ mkClientEnv manager url
```

Comment:
Pretty straightforward: fork (on green threads) `Async` processes ___. You can think of these as having a similar interface as JS Promises — it's a thing that may or may not be done yet. We _can_ block with `wait`, but since we want to react on the _first_ completion, we use `poll` which returns `Nothing` if it's still running

Reply:
The `Seconds` is a `maxTime`